can-stache.js

```diff
@@ -264,7 +264,9 @@ function stache (filename, template) {
 					// If we find a can-template we want to go back 2 in the stack to get it's inner content
 					// rather than the <can-template> element itself
 					var parent = state.sectionElementStack[state.sectionElementStack.length - 2];
-					parent.templates[oldNode.attrs.name] = makeRendererConvertScopes(renderer);
+					if (renderer) {// Only add the renderer if the template has content
+						parent.templates[oldNode.attrs.name] = makeRendererConvertScopes(renderer);
+					}
```

Comment:
Is it ok to not set the template property at all?

I see where this is used in can-component: https://github.com/canjs/can-component/blob/849f9056b9eacbf0bf57f634c94310198063e22f/can-component.js#L350-L353

Here it calls the function above: https://github.com/canjs/can-component/blob/849f9056b9eacbf0bf57f634c94310198063e22f/can-component.js#L80-L81

Reply:
Although `makeRendererConvertScopes` should maybe handle this internally, that might cause more problems than this solution. I would stick with what you have and just not set the property on `templates` if there is no renderer.